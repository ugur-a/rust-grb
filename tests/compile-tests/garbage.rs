use grb::prelude::*;
mod utils;

fn main() -> grb::Result<()> {
    create_model!(_g, m, x, y, z);

    c!(x + y + 2*z + 1 <= 0 in 1..0);
    c!(z + y in 0..z);
    c!(z + y in (x + 1)..);
    c!(y in x..z);

    Ok(())
}
