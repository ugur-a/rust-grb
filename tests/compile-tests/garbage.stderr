error[E0277]: the trait bound `Expr: From<bool>` is not satisfied
 --> tests/compile-tests/garbage.rs:7:8
  |
7 |     c!(x + y + 2*z + 1 <= 0 in 1..0);
  |        ^ the trait `From<bool>` is not implemented for `Expr`
  |
  = help: the following other types implement trait `From<T>`:
            <Expr as From<&T>>
            <Expr as From<LinExpr>>
            <Expr as From<QuadExpr>>
            <Expr as From<Var>>
            <Expr as From<f32>>
            <Expr as From<f64>>
            <Expr as From<i16>>
            <Expr as From<i32>>
          and 8 others

error[E0605]: non-primitive cast: `Var` as `f64`
 --> tests/compile-tests/garbage.rs:8:20
  |
8 |     c!(z + y in 0..z);
  |                    ^ an `as` expression can only be used to convert between primitive types or to coerce to a specific trait object

error[E0605]: non-primitive cast: `Var` as `f64`
  --> tests/compile-tests/garbage.rs:10:16
   |
10 |     c!(y in x..z);
   |                ^ an `as` expression can only be used to convert between primitive types or to coerce to a specific trait object

error[E0605]: non-primitive cast: `Var` as `f64`
  --> tests/compile-tests/garbage.rs:10:13
   |
10 |     c!(y in x..z);
   |             ^ an `as` expression can only be used to convert between primitive types or to coerce to a specific trait object
